### ISSUE TITLE
Support environment per channel

### ISSUE THREAD

**Issue by pcarranza** [gomeeseeks/meeseeks-box#107]
This would allow having different command configurations depending on the channel in which the job is being invoked, allowing to do things like invoking from a staging environment to do something, and from a production environment to do something else.

My gut feeling is that the environment should be stored in the database and that these environment values should be handled by admins.

thoughts @omame @ilya-f ?

**Comment by pcarranza** [comment]
Configuration, I like the idea of now leaking variables, and being able of keeping agents self-contained.

**Comment by ilya-f** [comment]
how do we intend to run multi-channel bot? if we only supply all the needed secrets for a channel and then containerize it, then its self-contained, but here we're talking about having something that holds all the keys for all channels eventually, right?

**Comment by pcarranza** [comment]
we can have the same bot in the multiple channels, we can also limit them to specific channels.

the thing is: given a bot with a command that is both valid for production and staging, the same command would be using a different endpoint, or auth token, depending on the channel.

what I'm thinking about is how this should be handled, but I'm also thinking about how multiple agents can be split and configured differently through the environment.

from a user perspective, what's the obvious thing that should happen when you issue a command? from here, how should things be configured?

**Comment by omame** [comment]
Can't we solve this problem by specifying a static label set in the bot configuration? So every time a user invokes a command the label set is always added to the invocation.
This would be trivial to code once the labels land.

**Comment by pcarranza** [comment]
so, something like appending labels to a command in a given channel to route to a specific agent?

that's an interesting idea.

would you hold this configuration in the database?

**Comment by omame** [comment]
Not necessarily. It could even be a simple flag to pass on the command line. Something like `-default-label-set='environment=staging,team=frontend'`.
It should be very static and only change when restarting the process.
